WordPress/gutenberg
# Comment pagination blocks: Add spacing support

## What?

Adds spacing block support to
Comments pagination, comments pagination next, previous and numbers.

## Why?

So that users can adjust the spacing, and for consistency.
Related: https://github.com/WordPress/gutenberg/issues/43243


## How?

Adds spacing block support in block.json.

## Testing Instructions

The testing steps are the same for the four blocks, test both individual blocks, global style settings, and theme.json style settings.

1. First make sure that your test installation has some comments, and that comment pagination is enabled in the WordPress admin area under Settings, Discussion, "Break comments into pages"
2. Create or open a post or page template in the site editor and add a comments block.
3. In the comments block, locate the nested comment pagination block.
4. Confirm if the comment pagination block and the inner blocks have a dimension panel with the margin and padding available, but hidden by default.
5. Make adjustments to the spacing. Confirm that the spacing is applied correctly in the editor.
6. Save and view the front, confirm if the spacing is applied correctly on the front.

Continue to test the block's layout (spacing) settings in the global styles sidebar in the site editor.
_Note that the comment pagination next/previous and numbers have the same names in the styles sidebar as the query pagination blocks. So if at first you fail, try the other._

Theme.json example code:
```
"styles": {
		"blocks": {
			"core/comments-pagination": {
				"spacing": {
					"padding" : {
							"top": "100px",
							"right": "100px",
							"bottom": "100px",
							"left": "100px"
						},
						"margin": {
							"top": "100px",
							"right": "100px",
							"bottom": "100px",
							"left": "100px"
						}
					}
			},
			"core/comments-pagination-next": {
				"spacing": {
					"padding": {
							"top": "100px",
							"right": "100px",
							"bottom": "100px",
							"left": "100px"
					},
					"margin": {
							"top": "100px",
							"right": "100px",
							"bottom": "100px",
							"left": "100px"
						}
					}
			},
			"core/comments-pagination-previous": {
				"spacing": {
					"padding" : {
							"top": "100px",
							"right": "100px",
							"bottom": "100px",
							"left": "100px"
						},
						"margin": {
							"top": "100px",
							"right": "100px",
							"bottom": "100px",
							"left": "100px"
						}
					}
			},
			"core/comments-pagination-numbers": {
				"spacing": {
					"padding" : {
							"top": "100px",
							"right": "100px",
							"bottom": "100px",
							"left": "100px"
						},
						"margin": {
							"top": "100px",
							"right": "100px",
							"bottom": "100px",
							"left": "100px"
						}
					}
			}
		}
	}
```

## Screenshots or screencast
